cli/submit.py

```diff
@@ -2,46 +2,45 @@
 import os
 
 import requests
-import yaml
+from ebi_eva_common_pyutils.config import WritableConfig
 from ebi_eva_common_pyutils.logger import logging_config
 
+from cli import SUB_CLI_CONFIG_FILE, __version__
 from cli.auth import get_auth
+from cli.docker_validator import READY_FOR_SUBMISSION_TO_EVA
 
 logger = logging_config.get_logger(__name__)
-SUB_CLI_CONFIG_FILE = ".eva-sub-cli-config.yml"
 SUB_CLI_CONFIG_KEY_SUBMISSION_ID = "submission_id"
 SUB_CLI_CONFIG_KEY_SUBMISSION_UPLOAD_URL = "submission_upload_url"
 SUBMISSION_INITIATE_URL = "http://www.ebi.ac.uk/eva/v1/submission/initiate"
 
 
 class StudySubmitter:
-    def __init__(self, submission_initiate_url=SUBMISSION_INITIATE_URL):
+    def __init__(self, submission_dir, submission_initiate_url=SUBMISSION_INITIATE_URL, submission_config: WritableConfig = None):
         self.auth = get_auth()
         self.submission_initiate_url = submission_initiate_url
-
-    def create_submission_config_file(self, submission_dir, submission_id, submission_upload_url):
-        submission_config_file = os.path.join(submission_dir, SUB_CLI_CONFIG_FILE)
-        config_data = {
-            SUB_CLI_CONFIG_KEY_SUBMISSION_ID: submission_id,
-            SUB_CLI_CONFIG_KEY_SUBMISSION_UPLOAD_URL: submission_upload_url
-        }
-        with open(submission_config_file, 'w') as open_file:
-            yaml.safe_dump(config_data, open_file)
-
-    def get_submission_id_and_upload_url(self, submission_dir):
-        submission_config_file = os.path.join(submission_dir, SUB_CLI_CONFIG_FILE)
-        if submission_config_file:
-            with (open(submission_config_file, 'r') as f):
-                submission_config_data = yaml.safe_load(f)
-                return submission_config_data[SUB_CLI_CONFIG_KEY_SUBMISSION_ID], submission_config_data[
-                    SUB_CLI_CONFIG_KEY_SUBMISSION_UPLOAD_URL]
+        if submission_config:
+            self.sub_config = submission_config
         else:
-            raise FileNotFoundError(f'Could not upload. No config file found for the submission in {submission_dir}.')
+            config_file = os.path.join(submission_dir, SUB_CLI_CONFIG_FILE)
+            self.sub_config = WritableConfig(config_file, version=__version__)
+
+    def update_config_with_submission_id_and_upload_url(self, submission_id, upload_url):
+        self.sub_config.set(SUB_CLI_CONFIG_KEY_SUBMISSION_ID, value=submission_id)
+        self.sub_config.set(SUB_CLI_CONFIG_KEY_SUBMISSION_UPLOAD_URL, value=upload_url)
+        self.sub_config.write()
```

Comment:
**Optional**: Since we have the ability to back up the existing config file, I am wondering if we might as well do it for troubleshooting purposes?

Reply:
updated